src/views/Form/index.tsx

```diff
@@ -1,47 +1,60 @@
 import React from 'react';
-import { useNavigate } from 'react-router-dom';
-import { MdCancel } from 'react-icons/md';
 
-import { Grid } from '@mui/material';
-import { Alert, Button } from 'shared-components';
-
-import TextInput from '../../components/TextInput';
-import Label from '../../components/Label';
+import {
+  Button,
+  TextInput,
+  InlineNotification,
+  FlexGrid,
+  Column,
+  Row,
+  Stack
+} from '@carbon/react';
+
+import LoadingButton from '../../components/LoadingButton';
 import UserTable from '../../components/UserTable';
 
 import SampleUser from '../../types/SampleUser';
 import getExceptionResponse from '../../service/GetExceptionResponse';
 
-import './styles.css';
 import { fetchApiRequest, createRequestInit } from '../../service/FetchApi';
-import CardLoader from '../../components/CardLoader';
+
+import './styles.css';
 
 const Form = () => {
   const BASE_URL = process.env.REACT_APP_SERVER_URL;
 
   const [firstName, setFirstName] = React.useState('');
   const [firstFeed, setFirstFeed] = React.useState('');
-  const [firstIsValid, setFirstIsValid] = React.useState<boolean | undefined>(undefined);
+  const [firstInvalid, setFirstInvalid] = React.useState<boolean | undefined>(undefined);
```

Comment:
Do we need a third state (one other than `true` or `false`, `undefined` in this case) here?

It feels like we're shoehorning our state to make it fit a boolean value, which should represent a state where there are only two answers _and_ these answers are either a variant of "yes" or a variant of "no".

Maybe "validity" isn't the way to go here: we consider an empty field as an invalid one, but if we need to differentiate an empty field from other invalid configurations, we must either use another concept (such as a `firstNameState` which can have one from a set of values, like `EMPTY`, `BELOW_MINIMUM`, `ABOVE_MAXIMUM` and `OK`) or split it into two or more states and work with their combinations.

[This article](https://note89.github.io/state-of-emergency/) might do a better job explaining my reservations than I can do here.

Reply:
Thanks for the tip (and great article BTW), we still need the third state to deal with the "empty but not wrong" scenario which happens when the page is loaded or the form is reset and the inputs are clear but the user hasn't interacted with them yet so we can't say it's wrong, but I created a type to deal with the set of values, it looks waaaaay better 🚀 .